dmutils/s3.py

```diff
@@ -138,6 +141,25 @@ def get_signed_url(self, path, expires_in=30):
                 ExpiresIn=expires_in,
             )
 
+    def get_full_signed_url(self, *args, assets_url=None, **kwargs):
+        """ Create a signed S3 document url, replacing its base url with `assets_url` if provided,
+            else will attempt to use DM_ASSETS_URL flask config parameter.
+
+            Otherwise arguments are the same as for `get_signed_url`
+        """
+        url = self.get_signed_url(*args, **kwargs)
+        if url is None:
+            return None
+
+        if assets_url is None:
+            if not has_app_context():
+                raise TypeError("assets_url argument must be provided if no app context is available")
+            assets_url = current_app.config["DM_ASSETS_URL"]
+
+        parsed_url = urlparse.urlparse(url)
+        parsed_assets_url = urlparse.urlparse(assets_url)
+        return parsed_url._replace(netloc=parsed_assets_url.netloc, scheme=parsed_assets_url.scheme).geturl()
```

Comment:
Does this duplicate what we have in `dmutils.documents.get_signed_url`? Or is that what the `self.get_signed_url` is using?

Reply:
As far as I can see, the other views in the Admin that fetch S3 docs use `dmutils.documents.get_signed_url`

Reply:
oh man i didn't even know about that...

Reply:
99% yes it does - seems the supplier FE doesn't know about that function either.